### PR TITLE
Apply Scala Syntax for code snippets

### PR DESCRIPTION
**Pull request by Varunram**
Intended as a fix for #2689. Changes primarily involve indentation changes (since using syntax highlighting with the existing indent would look a bit ugly). Do point out errors if any. Thanks!

### CODE REVIEW

**Comment by lforite** [comment]
I think we are good to go :-)

**Comment by OlivierBlanvillain** [comment]
Could you amend your commit message to make the first line < 72 characters long?

**Comment by Varunram** [comment]
Done! Off by one word :)

**Comment by OlivierBlanvillain** [comment]
It still conflicting :/

**Comment by Varunram** [comment]
Oh no, I didn't reset it. My bad. Just a moment.

**Comment by Varunram** [comment]
Done finally! Thanks a lot!

**Comment by felixmulder** [comment]
This guy needs a rebase before getting merged :)

I suggest doing something like this:

```bash
$ git remote add upstream git@github.com:lampepfl/dotty.git
$ git fetch upstream && git rebase -i upstream/master
```

This will ensure that you actually rebase against the latest master. Otherwise the problem might be that you're rebasing against an old HEAD.

**Comment by Varunram** [comment]
@felixmulder I think the conflicts are related to the changes. Weird that it picked out only two files though

File: docs/docs/reference/changed/implicit-conversions.md [hunk 1]
```diff
@@ -6,44 +6,45 @@ title: "Restrictions to Implicit Conversions"
...
+def convert[A, B](x: A)(implicit converter: ImplicitConverter[A, B]): B =
+  converter(x)
+```
 (In reality the Dotty compiler simulates the behavior of this method directly in
 its type checking because this turns out to be more efficient).
 
 In summary, previous code using implicit conversion parameters such as
-
+```scala
     def useConversion(implicit f: A => B) = {
```
Comment (OlivierBlanvillain) [inline docs/docs/reference/changed/implicit-conversions.md]: Indentation is off.

File: docs/docs/reference/changed/implicit-conversions.md [hunk 2]
```diff
@@ -6,44 +6,45 @@ title: "Restrictions to Implicit Conversions"
...
       val y: A = ...
       val x: B = a    // error under Dotty
     }
+```
 
 is no longer legal and has to be rewritten to
-
+```scala
     def useConversion(implicit f: ImplicitConverter[A, B]) = {
```
Comment (OlivierBlanvillain) [inline docs/docs/reference/changed/implicit-conversions.md]: Indentation is off.

File: docs/docs/reference/enums/desugarEnums.md [hunk 3]
```diff
@@ -31,122 +31,124 @@ into case classes, case objects or vals. Finally, rule (8) expands
...
+    case C <params> <body>
+    ```
+    expands according to two alternatives, depending whether `C` has type
+    parameters or not. If `C` has type parameters, they must have the same
+    names and appear in the same order as the enum type parameters `Ts`
+    (variances may be different, however). In this case
+    ```scala
+    case C [Ts] <params> <body>
+    ```
+      expands to
```
Comment (OlivierBlanvillain) [inline docs/docs/reference/enums/desugarEnums.md]: Two extra white spaces

**Comment by Varunram** [comment]
@OlivierBlanvillain sorry for the late reply. I think you're right! There must t be a better way to handle this. More info over at @2689

**Comment by OlivierBlanvillain** [comment]
Sounds good, thanks a lot for looking into this :smile:. I closing this PR for now, please reopen/open a new one if you have something new!

**Pull request closed without merging**